pkg/ioutil/delimiter-reader.go

```diff
@@ -0,0 +1,49 @@
+/*
+ * Minio Cloud Storage, (C) 2018 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ioutil
+
+import "io"
+import "bufio"
+
+var (
+	nByte byte = 10 // the byte that corresponds to the '\n' rune.
+)
+
+// Reader reduces the custom delimiter to `\n`.
+type Reader struct {
+	r         *bufio.Reader
+	delimiter rune
+}
+
+// DelimiterReader reads from the io.Reader and treats it as a buffer.
+func DelimiterReader(r io.Reader, delimiter rune) *Reader {
```

Comment:
Rename DelimiterReader to DelimitedReader

Reply:
Also can you add a test?

Reply:
yes will add

Reply:
Done